circle.yml

```diff
@@ -12,21 +12,16 @@ machine:
     - docker
 
 dependencies:
-  pre:
-    - eval $(node tasks/docker.js pull)
-  post:
-    - eval $(node tasks/docker.js run)
-    - npm run cibuild
+  override:
+    - npm install && npm dedupe && npm prune && npm install
+    - npm ls || true
+    - npm run docker -- pull
     - npm run pretest
-    - eval $(node tasks/docker.js setup)
-    - npm prune && npm ls
+    - npm run docker -- run
+    - npm run cibuild
+    - npm run docker -- setup
```

Comment:
Notice the docker commands are split between other command. This is reduce the risk of race conditions.